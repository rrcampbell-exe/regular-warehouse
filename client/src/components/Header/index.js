import React from "react";
import Auth from "../../utils/auth";
import { Link } from "react-router-dom";

const Header = () => {
  const logout = (event) => {
    event.preventDefault();
    Auth.logout();
  };

  return (
    <header className='bg-light'>
      <h1>ğŸ“¦ Regular Warehouse.</h1>
      <Link to="/" onClick={logout}>Log Out</Link>
    </header>
  );
};

export default Header;
