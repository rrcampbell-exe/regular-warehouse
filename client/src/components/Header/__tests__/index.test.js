import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);

describe('Header component', () => {
    it('renders', () => { // this does not pass (I think BrowserRouter fixed it?)
        render(<BrowserRouter><Header/></BrowserRouter>)
    })
    it('matches snapshot', () => {
        const { asFragment } = render(<Header/>); // does not pass
        expect(asFragment()).toMatchSnapshot();
    })
})