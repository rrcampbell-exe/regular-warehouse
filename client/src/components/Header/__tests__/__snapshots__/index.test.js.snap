// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-secondary"
  >
    <h1>
      ðŸ“¦ Regular Warehouse.
    </h1>
    <a
      href="/"
    >
      Log Out
    </a>
  </header>
</DocumentFragment>
`;
